man/fabric.7.md

```diff
@@ -238,6 +238,50 @@ FI_LOG_SUBSYS environment variables.
 - *mr*
 : Provides output specific to memory registration.
 
+# FILE SYSTEM CONFIGRATION INTERFACE
+
+Libfabric defines a set of file system interfaces that may be used to
+set global or provider specific parameters.  Conceptually, this interface
+is similar to the /proc interfaces used by ip.7 and tcp.7.
+Parameters can be accessed by reading or writing files in
+%SYSCONFDIR%/rdma/fabric.  The files are typically found at
+/etc/rdma/fabric or /usr/local/etc/rdma/fabric.
+The exact location of the files is controllable through build parameters,
+but for simplicity, the following documentation assumes that they are found
+under /etc/rdma/fabric.  The files do not exist by default and must be
+created by the administrator or Linux distro.  
+
+Parameter files located under /etc/rdma/fabric are treated as global and
+apply to all relevent providers.  Provider specific parameters may be set
+by writing to a file by the same name, but in a provider specific
+subdirectory.  For example, parameter files under /etc/rdma/fabric/verbs
+will apply only to the verbs provider.
+
+Note that providers may select alternate values that those
+specified.  This may be necessary in order to adhere to memory alignment
+restrictions, hardware maximums, or protocol limitations, for example.
+
+*conn_timeout - Connection Request Timeout (integer)*
+: Specifies the total time, in milliseconds, before a connection operation
+  times out.
+
+*def_inject_size - Default Inject Size (integer)*
```

Comment:
To your first paragraph - yes.  Provider A would fail, but provider B would succeed.  This is why we also allow per provider parameters.

When the values are read is a policy decision that we need to make.  The verbs provider, which currently uses these, reads them on an as need basis.  I.e. when fi_getinfo is called.  It does not cache the values, so changes could affect the running job.  This is related to my comment on determining the best way for the framework to assist here.  E.g. it can read the files and cache them if needed.